src/index.js

```diff
@@ -61,126 +66,197 @@ const monitorChromecastPlayer = function (player, options) {
+              });
+              break;
+          }
+          break;
+        case cast.framework.events.EventType.RATE_CHANGE:
+          player.mux.emit('ratechange');
+          break;
+        case cast.framework.events.EventType.TIME_UPDATE:
+          currentTime = event.currentMediaTime;
+          player.mux.emit('timeupdate');
+          break;
+        case cast.framework.events.EventType.SEGMENT_DOWNLOADED:
+          let now = Date.now();
+          let loadData = {
+            request_event_type: 'SEGMENT_DOWNLOADED',
+            request_start: now - event.downloadTime - 1,
+            request_response_start: now - event.downloadTime,
+            request_response_end: now,
+            request_bytes_loaded: event.size,
+            request_type: 'media'
+          };
+
+          player.mux.emit('requestcompleted', loadData);
+          break;
+
+        case cast.framework.events.EventType.BREAK_STARTED:
+          inAdBreak = true;
+          player.mux.emit('adbreakstart');
+          break;
+      }
+    }
+
+    if (inAdBreak === true) {
+      // chromecast will mix player messages with ad messages
+      // seperating out these ad events and supressing unneeded messages until the adbreak is over
+      switch (event.type) {
+        case cast.framework.events.EventType.BREAK_CLIP_LOADING:
+          if (adBreakClips !== null && event.breakClipId !== undefined) {
+            let breakClip = adBreakClips.find(o => o.id === event.breakClipId);
+            let adAssetUrl = (breakClip !== undefined && breakClip.contentUrl !== undefined) ? breakClip.contentUrl : (breakClip !== undefined && breakClip.contentId !== undefined) ? breakClip.contentId : null;
+
+            // Ad Tag URL is defined, but Chromecast does some weird auto-generated break clip stuff which doesn't carry the adTagUrl through to the generated breaks
+            // If google fix this, it can be added in the future
+            player.mux.emit('adplay', {
+              ad_asset_url: adAssetUrl
+            });
+          } else {
+            player.mux.emit('adplay');
+          }
+          break;
+        case cast.framework.events.EventType.PLAY: // cater for playback events within ad breaks, not captured by BREAK_ events
+          if (adPlaying) { player.mux.emit('adplay'); };
+          break;
+        case cast.framework.events.EventType.PAUSE: // cater for playback events within ad breaks, not captured by BREAK_ events
+          if (adPlaying) { player.mux.emit('adpause'); };
+          break;
+        case cast.framework.events.EventType.BREAK_CLIP_STARTED:
+          player.mux.emit('adplaying');
+          adPlaying = true;
+          break;
+        case cast.framework.events.EventType.PLAYING: // cater for playback events within ad breaks, not captured by BREAK_ events
+          if (adPlaying) { player.mux.emit('adplaying'); };
+          break;
+        case cast.framework.events.EventType.BREAK_CLIP_ENDED:
```

Comment:
Is this _always_ going to be fired in the cases of skips and other exits of ad playback? I'm tempted to be a little more careful here (or, rather, below) and set `adPlaying = false` both here and right before we set `inAdBreak = false`. Just extra careful, at not much cost.